src/mgr/InfoAction.cc

```diff
@@ -143,14 +143,17 @@ Mgr::InfoAction::dump(StoreEntry* entry)
     Must(entry != nullptr);
 
 #if XMALLOC_STATISTICS
+    // TODO: these should be moved to their own report
     if (UsingSmp())
-        storeAppendPrintf(entry, "by kid%d {\n", KidIdentifier);
+        storeAppendPrintf(entry, "---\nkid: %d\n", KidIdentifier);
     DumpMallocStatistics(entry);
     if (UsingSmp())
-        storeAppendPrintf(entry, "} by kid%d\n\n", KidIdentifier);
+        storeAppendPrintf(entry, "...\n");
 #endif
-    if (IamPrimaryProcess())
+    if (IamPrimaryProcess()) {
+        storeAppendPrintf(entry, "---\n");
```

Comment:
> Alex: I do insist on that encapsulation even if all bugs are fixed.

> Francesco: Working on it.

> Francesco: Now done.

Not done. This PR code is still (poorly) duplicated.

Reply:
> There is no way to not duplicate code here, except by factoring malloc statistics out

That assertion is false because there is another way: Factor out code that opens and closes reports (as originally suggested!). Call the right functions from the right places, eliminating (poor) code duplication (and possibly fixing other problems).

Reply:
See https://github.com/squid-cache/squid/pull/1784/files#r1588918114 for why the current "info" report is IMO broken, beyond what you write here.
Once that is fixed, this statement goes away because we will not have any more reports that are unaggregated and aggregated at the same time

Reply:
AFAICT, this line should be removed (along with PR-added braces). Removing it may not fix some problems in the official code (not sure), but _adding_ it creates problems in PR code.